iocore/nexthop/unit-tests/test_SharedExtendible.cc

```diff
@@ -0,0 +1,281 @@
+/** @file
+  Test file for SharedExtendible
+  @section license License
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+  Unless REQUIRE by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#include "catch.hpp"
+
+#include "SharedExtendible.h"
+#include <iostream>
+#include <string>
+#include <vector>
+#include <ctime>
+#include <thread>
+#include "ts/ink_atomic.h"
+
+using namespace std;
+
+// SharedExtendible is abstract and must be derived in a CRTP
+struct Derived : SharedExtendible<Derived> {
+  using Base_t = SharedExtendible<Derived>;
+  string m_str;
+};
+
+// something to store more complex than an int
+struct testField {
+  uint8_t arr[5];
+  static int alive;
+  testField()
+  {
+    uint8_t x = 1;
+    for (uint8_t &a : arr) {
+      a = x;
+      x *= 2;
+    }
+    alive++;
```

Comment:
`++alive` - use prefix unless there's a reason to use postfix. It's a good habit.